src/tests/nativeaot/SmokeTests/Reflection/Reflection.cs

```diff
@@ -1434,6 +1434,12 @@ public static void Run()
                     throw new Exception("GetValues");
             }
 
+            Console.WriteLine("Enum.GetEnumValuesAsUnderlyingType");
+            {
+                if (Enum.GetEnumValuesAsUnderlyingType(typeof(Mine)) is not int[])
```

Comment:
```suggestion
                if (Enum.GetEnumValuesAsUnderlyingType(typeof(Mine)).GetType() != typeof(int[]))
```
This is more strict check. `is not int[]` would pass even when the method returned `Mine[]`.